### ISSUE TITLE
add tests for sympy-bot

### ISSUE THREAD

**Issue by certik** [sympy/sympy-bot-old#80]
We should add tests for sympy-bot somehow, to test against regressions. Any ideas how to do that? I guess we should allow sympy-bot to use a user specified server, and the tests would specify to use some testing (local) server, and test that sympy-bot does what it is supposed to do.

**Comment by certik** [comment]
Yes, and this would also test the web application (at the same time?) somehow.

**Comment by MichaelMayorov** [comment]
I'm collecting ideas for my GSoC project about improving functionality and bug fixing work for sympy-bot.

May be that's old issue(4 months since open), but seems for me it's still actual. 
As I understand the main purpose of sympy-bot is reviewing incoming pull-requests, so why don't test this functional first?

I saw some scripts under "examples/' directory, they could be good start for creating tests. You told about user's local server, but may be we could just ship small repository with sympy-bot for checking pulls? Of course, It's not neccesary should be based on sympy codebase(at least sympy-bot could check any pull requests, not only related with sympy)

**Comment by certik** [comment]
I think it would be nice if sympy-bot could be easily adapted to test any other project, not just sympy.

The sympy-bot needs to be able to upload the appengine's reviews.sympy.org (see the `web` directory) and the API should be tested somehow. There are lots of things to be improved in the web interface.

**Comment by MichaelMayorov** [comment]
certik thanks, but what about:

> I saw some scripts under "examples/' directory, they could be good start for creating tests. You told about user's local server, but may be we could just ship small repository with sympy-bot for checking pulls? Of course, It's not neccesary should be based on sympy codebase(at least sympy-bot could check any pull requests, not only related with sympy)

**Comment by MichaelMayorov** [comment]
Sorry for posting here. Ondřej, could you review my ideas related with sympy bot https://groups.google.com/forum/?fromgroups#!topic/sympy/xDCOWh4d8Ug and write your thoughts about that?

**Comment by certik** [comment]
I will.

**Comment by asmeurer** [comment]
For Code-In, first, create a testing framework (one task), then add some tests (some more tasks).
